src/app/kvm/views/LXDClusterDetails/LXDClusterDetailsRedirect/LXDClusterDetailsRedirect.tsx

```diff
@@ -0,0 +1,45 @@
+import { Spinner } from "@canonical/react-components";
+import { useSelector } from "react-redux";
+import { Redirect } from "react-router-dom";
+
+import ModelNotFound from "app/base/components/ModelNotFound";
+import { useGetURLId } from "app/base/hooks/urls";
+import kvmURLs from "app/kvm/urls";
+import podSelectors from "app/store/pod/selectors";
+import { PodMeta } from "app/store/pod/types";
+import type { RootState } from "app/store/root/types";
+import type { VMCluster } from "app/store/vmcluster/types";
+import { isId } from "app/utils";
+
+type Props = {
+  clusterId: VMCluster["id"];
+};
+
+export enum Label {
+  Loading = "Loading LXD host",
+}
+
+const LXDClusterDetailsRedirect = ({ clusterId }: Props): JSX.Element => {
+  const hostId = useGetURLId(PodMeta.PK, "hostId");
+  const host = useSelector((state: RootState) =>
+    podSelectors.getById(state, hostId)
+  );
+  const hostsLoaded = useSelector(podSelectors.loaded);
+  if (!hostsLoaded) {
+    return <Spinner aria-label={Label.Loading} />;
+  }
+  if (!isId(hostId) || !host) {
+    return (
+      <ModelNotFound
+        id={hostId}
+        inSection={false}
+        linkText="View all LXD hosts in this cluster"
+        linkURL={kvmURLs.lxd.cluster.hosts({ clusterId })}
+        modelName="LXD host"
+      />
+    );
+  }
+  return <Redirect to={kvmURLs.lxd.cluster.host.edit({ clusterId, hostId })} />;
```

Comment:
This redirect had to be moved into its own file as it needed to be under a route for `useParams` to work.